### DISCUSSION TITLE
Error in SUDFIFO register when configuring tinyUSB with MAX3421E device

### DISCUSSION THREAD

**Discussion by jvelazquez98** [hathach/tinyusb#2874] [Q&A]
Hi,

I'm facing an issue with tinyUSB when trying to enumerate my USB devices using the MAX3421E.

- The initialization procedure is done apparently good, and the USB device is properly detected over the D+ and D- level change.

- Then, the host tries to start the enumeration procedure by writing 8 bytes on the SUDFIFO register (specifically, 0x80 0x06 0x00 0x01 0x00 0x00 0x08 0x00).

- After that, the HXFR register is set, instructing the SIE from MAX3421E to start a SETUP packet transmission.

- The SETUP packet is indeed started, but after it, a DATA0 transmission contaning the data previously stored in the SUDFIFO register should be written. However, what I'm seeing is a DATA0 transmission with all 8 bytes set 0x00.

Is there any configuration out of the standard one that I'm missing? Is there any particular SPI speed and/or parameters that should be taken into account when writing onto FIFO registers?

Note that this behavior is happening both with USB hubs and with normal USB devices.

Hope that you can find any potential reason for this issue.

Thanks in advance and best regards,
Jaime

**Comment by hathach** [comment]
what is your setup
- board, mcu
- which example you are running
- which device you are testing with
